engine/runtime/src/main/java/org/enso/interpreter/runtime/warning/AppendWarningNode.java

```diff
@@ -94,12 +96,58 @@ WithWarnings doMultipleWarningsArray(
     return new WithWarnings(value, warnsLimit, isLimitReached, warnsMap);
   }
 
+  /**
+   * This specialization should be the most frequent - just wrapping the given {@code object} with
+   * warnings hash map
+   */
+  @Specialization(guards = {"!isWithWarns(object)"})
+  WithWarnings doObjectMultipleWarningsHashMap(
+      VirtualFrame frame, Object object, EnsoHashMap newWarnsMap) {
+    assert !(object instanceof WithWarnings);
+    int warnsLimit = EnsoContext.get(this).getWarningsLimit();
+    return new WithWarnings(
+        object, warnsLimit, newWarnsMap.getHashSize() >= warnsLimit, newWarnsMap);
```

Comment:
Can someone instantiate new `WithWarnings` with a different limit than `EnsoContext.get(this).getWarningsLimit()`? If not, that we could hide all this limit manipulation inside of the `WithWarnings` class.

Reply:
So far, only [SortVectorNode](https://github.com/enso-org/enso/blob/788115f79311068d2f5eb774f93672a4de207178/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/SortVectorNode.java#L359) uses a different warning limit than the one stored in the `EnsoContext`.